src/shared/components/case-viewer/case-full-access-view/case-full-access-view.component.ts

```diff
@@ -250,12 +255,26 @@ export class CaseFullAccessViewComponent implements OnInit, OnDestroy, AfterView
   }
 
   public onLinkClicked(triggerOutputEventText: string): void {
-    const matTab = this.tabGroup._tabs.find((x) => x.textLabel === triggerOutputEventText);
-    if (matTab && matTab.position) {
-      this.tabGroup.selectedIndex = matTab.position;
+    // Get the *absolute* (not relative) index of the target tab and set as the active tab, using the selectedIndex input
+    // of mat-tab-group (bound to selectedTabIndex)
+    const targetTabIndex = this.tabGroup._tabs.toArray().findIndex(tab => tab.textLabel === triggerOutputEventText);
+    if (targetTabIndex > -1) {
+      this.selectedTabIndex = targetTabIndex;
     }
   }
 
+  public getCaseFlagsTabName(): string {
+    // Determine which tab contains the FlagLauncher CaseField type, from the CaseView object in the snapshot data,
+    // and get its label
+    if (!this.caseFlagsTabName && this.route.snapshot.data.case && this.route.snapshot.data.case.tabs) {
+      this.caseFlagsTabName = (this.route.snapshot.data.case.tabs as CaseTab[])
+      .filter(tab => tab.fields && tab.fields
+        .some(caseField => caseField.field_type.id === 'FlagLauncher' && caseField.field_type.type === 'FlagLauncher'))[0].label;
```

Comment:
Is that field_type.id === 'FlagLauncher' enough to get the filed_type which is 'FlagLauncher'? Do we need to judge that again from the caseField.field_type.type which seems duplicated, does it have this case when that id === 'FlagLauncher' but type is different?

Reply:
hi @rayliangatos - I have checked a bit further and it does seem there are cases where `field_type.id` and `field_type.type` are not the same. Examples include collections and Complex types. I will change the check to look at `field_type.type` only.